test/test_data_frame_variable_operation.rb

```diff
@@ -439,6 +439,14 @@ class DataFrameVariableOperationTest < Test::Unit::TestCase
       assert_equal str2, @df2.assign { assigner2.to_a }.tdr_str
     end
 
+    sub_test_case 'Dataframe with zero n_records' do
+      test 'assign by block preserves types' do
+        assert_equal :double, @df.b.type
+        result_df = @df.filter(@df.a == 10).assign(:b) { b.multiply(1)}
```

Comment:
How about using `@df.c == "nonexistent"` or adding a comment why we use `10` here?